test/org/opensolaris/opengrok/condition/ConditionalRunRule.java

```diff
@@ -64,34 +68,70 @@ public Statement apply(Statement aStatement, Description aDescription) {
-        return aDescription.getTestClass().getAnnotation(ConditionalRun.class) != null;
+        return aDescription.getTestClass().getAnnotationsByType(ConditionalRun.class).length > 0;
     }
 
     private static RunCondition getIgnoreConditionOnClass(Description aDescription) {
-        ConditionalRun annotation = aDescription.getTestClass().getAnnotation(ConditionalRun.class);
-        return new IgnoreConditionCreator(aDescription.getTestClass(), annotation).create();
+        ConditionalRun[] annotations = aDescription.getTestClass().getAnnotationsByType(ConditionalRun.class);
+        return new IgnoreConditionCreator(aDescription.getTestClass(), annotations).create();
     }
 
     private static boolean hasConditionalIgnoreAnnotationOnMethod(Description aDescription) {
-        return aDescription.getAnnotation(ConditionalRun.class) != null;
+        try {
+            // this is possible because test methods must not have any argument
+            Method testMethod = aDescription.getTestClass().getMethod(aDescription.getMethodName());
+            return testMethod.getAnnotationsByType(ConditionalRun.class).length > 0;
+        } catch (NoSuchMethodException | SecurityException ex) {
+            throw new RuntimeException(ex);
+        }
     }
 
     private static RunCondition getIgnoreConditionOnMethod(Description aDescription) {
-        ConditionalRun annotation = aDescription.getAnnotation(ConditionalRun.class);
-        return new IgnoreConditionCreator(aDescription.getTestClass(), annotation).create();
+        try {
+            // this is possible because test methods must not have any argument
+            ConditionalRun[] annotations = aDescription.getTestClass().getMethod(aDescription.getMethodName()).getAnnotationsByType(ConditionalRun.class);
+            return new IgnoreConditionCreator(aDescription.getTestClass(), annotations).create();
+        } catch (NoSuchMethodException | SecurityException ex) {
+            throw new RuntimeException(ex);
+        }
     }
 
-    private static class IgnoreConditionCreator {
+    /**
+     * Container for several conditions joined by an AND operator.
+     */
+    protected static class CompositeCondition implements RunCondition {
+
+        List<RunCondition> conditions = new LinkedList<>();
+
+        public boolean add(RunCondition e) {
+            return conditions.add(e);
+        }
+
+        @Override
+        public boolean isSatisfied() {
+            for (RunCondition condition : conditions) {
+                if (!condition.isSatisfied()) {
+                    return false;
+                }
+            }
+            return true;
+        }
+    }
+
+    protected static class IgnoreConditionCreator {
 
         private final Class<?> mTestClass;
-        private final Class<? extends RunCondition> conditionType;
+        private final List<Class<? extends RunCondition>> conditionType;
```

Comment:
rename to signify it is now holding more conditions ?

Reply:
fixed